be/btree.c

```diff
@@ -520,19 +520,19 @@ static void be_btree_split_child(struct m0_be_btree *btree,
 	for (i = parent->bt_num_active_key + 1; i > index + 1; i--) {
 		parent->bt_child_arr[i] = parent->bt_child_arr[i - 1];
 		parent->bt_kv_arr[i - 1] = parent->bt_kv_arr[i - 2];
-		parent->bt_ik[i - 1] = parent->bt_ik[i - 2];
-		parent->allocated[i - 1] = parent->allocated[i - 2];
-		if(btree->bb_ops->ko_type != M0_BBT_CAS_CTG)
-		{
-			parent->bt_kv_arr[i - 1].btree_key = &parent->bt_ik[i - 1].inlkey;
-		}
+		// parent->bt_ik[i - 1] = parent->bt_ik[i - 2];
+		// parent->allocated[i - 1] = parent->allocated[i - 2];
+		// if(btree->bb_ops->ko_type != M0_BBT_CAS_CTG)
+		// {
+		// 	parent->bt_kv_arr[i - 1].btree_key = &parent->bt_kv_arr[i - 1].inlkey;
+		// }
 	}
 
 	/*  Update parent */
 	parent->bt_child_arr[index + 1] = new_child;
 	parent->bt_kv_arr[index] = child->bt_kv_arr[BTREE_FAN_OUT - 1];
-	parent->bt_ik[index] = child->bt_ik[BTREE_FAN_OUT - 1];
-	parent->allocated[index] = child->allocated[BTREE_FAN_OUT - 1];
+	*parent->bt_ik[index].inlkey = *((char *)child->bt_kv_arr[BTREE_FAN_OUT - 1].btree_key);
```

Comment:
Here also you need to do memcpy. Also you cannot use 'index' for bt_ik you need to find the not-allocated entry to use.